src/mito2/src/manifest/action.rs

```diff
@@ -196,8 +260,55 @@ impl RegionManifestBuilder {
     }
 }
 
+/// A record of removed files in the region manifest.
+/// This is used to keep track of files that have been removed from the manifest but may still
+/// be kept for a while
+#[derive(Serialize, Deserialize, Clone, Debug, Default, PartialEq, Eq)]
+pub struct RemovedFilesRecord {
+    /// a list of `(FileIds, timestamp)` pairs, where the timestamp is the time when
+    /// the files are removed from manifest. The timestamp is in milliseconds since unix epoch.
+    pub removed_files: Vec<(HashSet<FileId>, i64)>,
```

Comment:
Can we avoid using tuples here? It makes the deserialized JSON unclear. Use a struct to replace the tuple `(HashSet<FileId>, i64)`.
The RemovedFilesRecord itself could be replaced by a Vec.